mlflow/genai/judges/optimizers/dspy_utils.py

```diff
@@ -24,6 +26,73 @@
 _logger = logging.getLogger(__name__)
 
 
+def _to_attrdict(obj):
+    """Recursively convert nested dicts/lists to AttrDicts."""
+    if isinstance(obj, dict):
+        return AttrDict({k: _to_attrdict(v) for k, v in obj.items()})
+    elif isinstance(obj, list):
+        return [_to_attrdict(item) for item in obj]
+    else:
+        return obj
+
+
+def _call_managed_rag_client(user_prompt: str, system_prompt: str | None = None):
```

Comment:
Ben might have already introduced this into another utility file. Not sure but worth checking.